frontend/app/routes/_public+/intake+/$id+/communication-preference.tsx

```diff
@@ -26,46 +28,60 @@ export const handle = {
+  const { state } = await intakeFlow.loadState({ request, params });
+
+  return json({ COMMUNICATION_METHOD_DIGITAL_ID, preferredLanguages, preferredCommunicationMethods, state: state.communicationPreferences });
 }
 
-export async function action({ request }: ActionFunctionArgs) {
-  const formDataSchema = z.object({
-    preferredLanguage: z.enum(['en', 'fr']),
-  });
+export async function action({ request, params }: ActionFunctionArgs) {
+  const intakeFlow = getIntakeFlow();
+  const { id } = await intakeFlow.loadState({ request, params });
+
+  const { COMMUNICATION_METHOD_DIGITAL_ID } = getClientEnv();
+
+  const formSchema = z
+    .object({
+      preferredMethod: z.string({ required_error: 'empty-radio' }),
+      email: z.string().min(1, { message: 'empty-field' }).email({ message: 'invalid-email' }).optional(),
+      confirmEmail: z.string().min(1, { message: 'empty-field' }).optional(),
+      preferredLanguage: z.string({ required_error: 'empty-radio' }),
+    })
+    .refine((allValues) => allValues.preferredMethod != COMMUNICATION_METHOD_DIGITAL_ID || (allValues.preferredMethod === COMMUNICATION_METHOD_DIGITAL_ID && allValues.email?.trim() === allValues.confirmEmail?.trim()), { message: 'not-the-same' });
 
   const formData = Object.fromEntries(await request.formData());
-  const parsedDataResult = formDataSchema.safeParse(formData);
+  const parsedDataResult = formSchema.safeParse(formData);
+
   if (!parsedDataResult.success) {
     return json({
       errors: parsedDataResult.error.format(),
-      formData: formData as Partial<z.infer<typeof formDataSchema>>,
+      formData: formData as Partial<z.infer<typeof formSchema>>,
     });
   }
 
-  const sessionService = await getSessionService();
-  const session = await sessionService.getSession(request);
-  session.set('newPreferredLanguage', parsedDataResult.data.preferredLanguage);
-
-  return redirect('/personal-information/preferred-language/confirm', {
-    headers: {
-      'Set-Cookie': await sessionService.commitSession(session),
-    },
+  const sessionResponseInit = await intakeFlow.saveState({
+    request,
+    params,
+    state: { communicationPreferences: parsedDataResult.data },
   });
+
+  return redirect(`/intake/${id}/personal-info`, sessionResponseInit);
 }
 
-export default function CommunicationPreferencePage() {
-  const { COMMUNICATION_METHOD_DIGITAL_ID, preferredLanguages, preferredCommunicationMethods } = useLoaderData<typeof loader>();
+export default function IntakeFlowCommunicationPreferencePage() {
+  const { COMMUNICATION_METHOD_DIGITAL_ID, preferredLanguages, preferredCommunicationMethods, state } = useLoaderData<typeof loader>();
   const { i18n, t } = useTranslation(i18nNamespaces);
-  const [isDigitalMethodChecked, setDigitalMethodChecked] = useState(false);
+  const [isDigitalMethodChecked, setDigitalMethodChecked] = useState(state && state.preferredMethod === COMMUNICATION_METHOD_DIGITAL_ID ? true : false);
```

Comment:
```suggestion
  const [isDigitalMethodChecked, setDigitalMethodChecked] = useState(state?.preferredMethod === COMMUNICATION_METHOD_DIGITAL_ID);
```

Reply:
The return value of "useState" should be destructured and named symmetrically. [S6754](https://sonarsource.github.io/rspec/#/rspec/S6754)

Reply:
fixed

Reply:
You currently have

``` typescript
const [digitalMethodChecked, setDigitalMethodChecked] = useState(state && state.preferredMethod === digitalCommunicationMethodId);
```

But I think you can remove the first `state` check and do this:

``` typescript
const [digitalMethodChecked, setDigitalMethodChecked] = useState(state?.preferredMethod === digitalCommunicationMethodId);
```

(am I wrong about that?)